src/sim/plotting.py

```diff
@@ -376,5 +376,164 @@ def policy_plot(self, qEs, risks, save_path=None):
+
+        ax.set_xlabel('X position')
+        ax.set_ylabel('Y position')
+        ax.set_title('Vehicle Trajectories')
+        ax.legend()
+
+        plt.tight_layout()
+
+        if save_path:
+            plt.savefig(save_path) # results/plot
+
+        return fig, ax
+
+    
+    def speed_plot(self, fig=None, ax=None, save_path="results/plots/speed_plot.png", plot_kwargs: Optional[dict] = None):
+        """
+        Plot the speed of vehicles over time a specific duration.
+        """
+        plot_kwargs = plot_kwargs or {}
+
+        if ax is None:
+            plot_kwargs['figsize'] = plot_kwargs.get('figsize', (10, 6)) # Random Size
+            fig, ax = plt.subplots(**plot_kwargs)
+
+        vehicles = self.ds.vehicles
+        for vehicle in vehicles:
+            speed = self.ds.speed.sel(vehicle=vehicle)
+            time = self.ds.time
+            ax.plot(time, speed, label=f'Vehicle {vehicle.item()}')
+
+        ax.set_xlabel('Time')
+        ax.set_ylabel('Speed')
+        ax.set_title('Vehicle Speeds Over Time')
+        ax.legend()
+
+        plt.tight_layout()
+
+        if save_path:
+            plt.savefig(save_path)
+
+        return fig, ax
+
+    def lane_occupancy_plot(self, fig=None, ax=None, save_path="results/plots/lane_occupancy_plot.png", plot_kwargs: Optional[dict] = None):
+        """
+        Plot the lane occupancy over time
+        """
+        plot_kwargs = plot_kwargs or {}
+
+        if ax is None:
+            plot_kwargs['figsize'] = plot_kwargs.get('figsize', (10, 6)) # Random Size
+            fig, ax = plt.subplots(**plot_kwargs)
+
+        lane_occupancy = self.ds.lane_occupancy
+        time = self.ds.time
+        lanes = self.ds.lanes
+
+        for lane in lanes:
+            occupancy = lane_occupancy.sel(lane=lane)
+            ax.plot(time, occupancy, label=f'Lane {lane.item()}')
+
+        ax.set_xlabel('Time')
+        ax.set_ylabel('Occupancy')
+        ax.set_title('Lane Occupancy Over Time')
+        ax.legend()
+
+        plt.tight_layout()
+
+        if save_path:
+            plt.savefig(save_path)
+
+        return fig, ax
+
+    def safety_distance_plot(self, fig=None, ax=None, save_path="results/plots/safety_distance_plot.png", plot_kwargs: Optional[dict] = None):
+        """
+        Plot the safety distance between vehicles over time
+        """
+        plot_kwargs = plot_kwargs or {}
+
+        if ax is None:
+            plot_kwargs['figsize'] = plot_kwargs.get('figsize', (10, 6)) # Random Size
+            fig, ax = plt.subplots(**plot_kwargs)
+
+        vehicles = self.ds.vehicles
+        time = self.ds.time
+        
+        for i in range(len(vehicles) - 1):
+            vehicle1 = vehicles[i]
+            vehicle2 = vehicles[i + 1]
+            distance = self.ds.distance_between.sel(vehicle1=vehicle1, vehicle2=vehicle2)
+            ax.plot(time, distance, label=f'Between Vehicle {vehicle1.item()} and {vehicle2.item()}')
```

Comment:
This assumes some sort of ordering and that there is only a single line of cars etc.
I don't think this plot is very functional as is. 
Vehicle density (ie congestion) could be helpful -- but maybe that is what lane occupancy is going for? That being said, I think congestion (spanning across all lanes in a given AVRoad) is more interesting than per-lane occupancy.

I think we should scrap this plot and change lane occupancy to congestion.

Reply:
Please check